File: debug_airtable.js


import 'dotenv/config';

const AIRTABLE_PAT = process.env.AIRTABLE_PAT || '';
const BASE_ID = 'appXUeeWN1uD9NdCW';
const USERS_TABLE = 'tblGWh6XpdEZxw8AE';
const PLANS_TABLE = 'tblq2T5OUyrDFCda9';
const LOGS_TABLE_QUESTIONS = 'tblvHx1DzwuTq3TJE';

async function queryAirtable(tableId, filterByFormula) {
    const url = `https://api.airtable.com/v0/${BASE_ID}/${tableId}?filterByFormula=${encodeURIComponent(filterByFormula)}`;
    console.log(`Querying table ${tableId} with formula: ${filterByFormula}`);
    const res = await fetch(url, {
        headers: { Authorization: `Bearer ${AIRTABLE_PAT}` }
    });
    if (!res.ok) {
        const errorBody = await res.text();
        console.error(`Error (${res.status}): ${errorBody}`);
        return;
    }
    const data = await res.json();
    console.log(`Success! Found ${data.records.length} records.`);
}

async function run() {
    console.log("--- Testing Users Table (Email) ---");
    await queryAirtable(USERS_TABLE, "{email} = 'piers.fawkes@psfk.com'");

    console.log("\n--- Testing Plans Table (Name) ---");
    await queryAirtable(PLANS_TABLE, "{Name} = 'Free'");

    console.log("\n--- Testing Users Table (User Name) ---");
    await queryAirtable(USERS_TABLE, "{User Name} = 'testuser'");

    console.log("\n--- Testing Plans Table (Plan Name) ---");
    await queryAirtable(PLANS_TABLE, "{Plan Name} = 'Free'");

    console.log("\n--- Testing Users Table (name - lowercase) ---");
    await queryAirtable(USERS_TABLE, "{name} = 'test'");
    console.log("\n--- Testing WRITE Permission (Create Log) ---");
    const logUrl = `https://api.airtable.com/v0/${BASE_ID}/${LOGS_TABLE_QUESTIONS}`;
    const logData = {
        fields: {
            "question": "DEBUG_TEST_WRITE",
            "userEmail": "debug@test.com",
            "Date": new Date().toISOString()
        }
    };
    try {
        const res = await fetch(logUrl, {
            method: 'POST',
            headers: {
                'Authorization': `Bearer ${AIRTABLE_PAT}`,
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(logData)
        });
        if (!res.ok) {
            console.error(`WRITE Failed (${res.status}): ${await res.text()}`);
        } else {
            const json = await res.json();
            console.log("WRITE Success! Created record:", json.id);
            // Cleanup?
        }
    } catch (e) {
        console.error("WRITE Error:", e);
    }
}

run();

File: debug_airtable.py

import urllib.request
import urllib.parse
import json
import ssl

import os
AIRTABLE_PAT = os.environ.get('AIRTABLE_PAT', '')
BASE_ID = 'appXUeeWN1uD9NdCW'
USERS_TABLE = 'tblGWh6XpdEZxw8AE'
PLANS_TABLE = 'tblq2T5OUyrDFCda9'
ACCOUNTS_TABLE = 'tblt6mh0XQOablFDX'
API_KEYS_TABLE = 'tblsDGYv8pFpNegcf'

def query_airtable(table_id, formula):
    encoded_formula = urllib.parse.quote(formula)
    url = f"https://api.airtable.com/v0/{BASE_ID}/{table_id}?filterByFormula={encoded_formula}"
    print(f"Querying table {table_id} with formula: {formula}")
    
    req = urllib.request.Request(url)
    req.add_header('Authorization', f'Bearer {AIRTABLE_PAT}')
    
    context = ssl._create_unverified_context()
    
    try:
        with urllib.request.urlopen(req, context=context) as response:
            data = json.loads(response.read().decode('utf-8'))
            print(f"Success! Found {len(data.get('records', []))} records.")
    except urllib.error.HTTPError as e:
        error_body = e.read().decode('utf-8')
        print(f"Error ({e.code}): {error_body}")
    except Exception as e:
        print(f"Exception: {str(e)}")




print("\n--- Inspecting ACCOUNTS_TABLE fields ---")
try:
    url = f"https://api.airtable.com/v0/{BASE_ID}/{ACCOUNTS_TABLE}?maxRecords=1"
    req = urllib.request.Request(url)
    req.add_header('Authorization', f'Bearer {AIRTABLE_PAT}')
    context = ssl._create_unverified_context()
    with urllib.request.urlopen(req, context=context) as response:
        data = json.loads(response.read().decode('utf-8'))
        if data.get('records'):
            print("Fields found:", list(data['records'][0]['fields'].keys()))
        else:
            print("No records found in ACCOUNTS_TABLE")
except Exception as e:
    print(f"Inspection Error: {e}")

print("\n--- Inspecting API_KEYS_TABLE fields ---")
try:
    url = f"https://api.airtable.com/v0/{BASE_ID}/{API_KEYS_TABLE}?maxRecords=1"
    req = urllib.request.Request(url)
    req.add_header('Authorization', f'Bearer {AIRTABLE_PAT}')
    context = ssl._create_unverified_context()
    with urllib.request.urlopen(req, context=context) as response:
        data = json.loads(response.read().decode('utf-8'))
        if data.get('records'):
            print("Fields found:", list(data['records'][0]['fields'].keys()))
        else:
            print("No records found in API_KEYS_TABLE")
except Exception as e:
    print(f"Inspection Error: {e}")



